ModConfigMenu/ModConfigMenu/Src/ModConfigMenuAPI/MCM_API_Includes.uci

```diff
@@ -0,0 +1,44 @@
+`define MCM_MAJOR_VERSION 0
+`define MCM_MINOR_VERSION 2
+
+`define CheckVersionAndRegister(screenInst, handler)\
+    if (MCM_API(`{screenInst}) != none){MCM_API(`{screenInst}).RegisterClientMod(`{MCM_MAJOR_VERSION}, `{MCM_MINOR_VERSION}, `{handler});}
+
+`define MCM_API_Boilerplate(handler)\
+	event OnInit(UIScreen Screen)\
+	{\
+		`CheckVersionAndRegister(Screen, `handler);\
+	}\
+	defaultproperties\
+	{\
+		ScreenClass = class'MCM_OptionsScreen';\
+	}
+
+`define GenerateSettingName(Varname) `{Varname}_BasicSetting
+`define GenerateHandlerName(Varname) `{Varname}_BasicHandler
+
+`define BasicLabel(Page, Varname, Label, Tooltip)\
+	`{Page}.AddLabel(`GenerateSettingName(`Varname), `Label, `Tooltip);
+`define BasicButton(Page, Varname, Label, Tooltip, ButtonLabel)\
+	`{Page}.AddButton(`GenerateSettingName(`Varname), `Label, `Tooltip, `ButtonLabel, `GenerateHandlerName(`Varname));
+`define BasicCheckbox(Page, Varname, Label, Tooltip)\
+	`{Page}.AddCheckbox(`GenerateSettingName(`Varname), `Label, `Tooltip, `Varname, `GenerateHandlerName(`Varname));
+`define BasicSlider(Page, Varname, Label, Tooltip, SliderMin, SliderMax, SliderStep)\
+	`{Page}.AddSlider(`GenerateSettingName(`Varname), `Label, `Tooltip, `SliderMin, `SliderMax, `SliderStep, `Varname, `GenerateHandlerName(`Varname));
+`define BasicSpinner(Page, Varname, Label, Tooltip, Options)\
+	`{Page}.AddSpinner(`GenerateSettingName(`Varname), `Label, `Tooltip, `Options, `Varname, `GenerateHandlerName(`Varname));
+`define BasicDropdown(Page, Varname, Label, Tooltip, Options)\
+	`{Page}.AddDropdown(`GenerateSettingName(`Varname), `Label, `Tooltip, `Options, `Varname, `GenerateHandlerName(`Varname));
+
+// Button handler works a bit differently because you want to implement the handler yourself.
+`define BasicButtonHandler(Varname)\
+	function `GenerateHandlerName(`Varname)(MCM_API_Setting Setting, name SettingName)
+
+`define BasicCheckboxHandler(Varname)\
+	function `GenerateHandlerName(`Varname)(MCM_API_Setting Setting, name SettingName, bool SettingValue) {`Varname = SettingValue;}
+`define BasicSliderHandler(Varname)\
+	function `GenerateHandlerName(`Varname)(MCM_API_Setting Setting, name SettingName, float SettingValue) {`Varname = SettingValue;}
+`define BasicSpinnerHandler(Varname)\
+	function `GenerateHandlerName(`Varname)(MCM_API_Setting Setting, name SettingName, string SettingValue) {`Varname = SettingValue;}
+`define BasicDropdownHandler(Varname)\
+	function `GenerateHandlerName(`Varname)(MCM_API_Setting Setting, name SettingName, string SettingValue) {`Varname = SettingValue;}

```

Comment:
We should preface all the macros with the same string, since there's no namespaces or anything.  Something like `MCM_DropdownHandler`